### ISSUE TITLE
Very small ggiraph plots when opening shiny app in browser

### ISSUE THREAD

**Issue by antuki** [davidgohel/ggiraph#74]
Hello, 

I'm developping a shiny app with ggiraph plots inside and i've noticed that when i'm opening the shiny app in browser, the ggiraph plots are very very very small. I can't understand why and didn't succeed in making them bigger... 

Do you have a solution for me ? 

I reproduced a simple example and made screen shots. Hope it will help

Thanks a lot...

```
library(shiny)
library(ggiraph)

shinyApp(
  ui = fluidPage(
    sidebarLayout(
      sidebarPanel(width=3,
                   h6("Hello")
      ),
      mainPanel(width=9,
                tabsetPanel(
                  tabPanel("PLOT",
                           ggiraphOutput("plot",width = "100%", height = "100%")
                  )
                )
      )
    )
  ),
  server = function(input, output, session) { 
    my_gg <- ggplot(mpg, aes( x = displ, y = cty, color = hwy) ) + geom_point_interactive(aes(tooltip = model), size = 2) 
    output$plot <- renderggiraph({
     ggiraph(code = print(my_gg))
     })  
  }
)
```

```
> sessionInfo()
R version 3.3.2 (2016-10-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=French_France.1252  LC_CTYPE=French_France.1252    LC_MONETARY=French_France.1252
[4] LC_NUMERIC=C                   LC_TIME=French_France.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gdtools_0.1.6   bindrcpp_0.2    ggiraph_0.4.1   dplyr_0.7.2     purrr_0.2.3     readr_1.1.1    
 [7] tidyr_0.7.1     tibble_1.3.4    ggplot2_2.2.1   tidyverse_1.1.1 haven_1.1.0     shiny_1.0.5    

loaded via a namespace (and not attached):
 [1] reshape2_1.4.2    lattice_0.20-34   colorspace_1.3-2  htmltools_0.3.6   yaml_2.1.14      
 [6] base64enc_0.1-3   rlang_0.1.2       R.oo_1.21.0       R.utils_2.5.0     foreign_0.8-67   
[11] glue_1.1.1        uuid_0.1-2        modelr_0.1.1      readxl_1.0.0      bindr_0.1        
[16] plyr_1.8.4        stringr_1.2.0     munsell_0.4.3     gtable_0.2.0      cellranger_1.1.0 
[21] zip_1.0.0         R.methodsS3_1.7.1 rvest_0.3.2       htmlwidgets_0.9   psych_1.7.5      
[26] labeling_0.3      forcats_0.2.0     httpuv_1.3.5      parallel_3.3.2    broom_0.4.2      
[31] Rcpp_0.12.12      xtable_1.8-2      DT_0.2            scales_0.5.0      jsonlite_1.5     
[36] mime_0.5          mnormt_1.5-5      hms_0.3           digest_0.6.12     stringi_1.1.5    
[41] grid_3.3.2        tools_3.3.2       magrittr_1.5      lazyeval_0.2.0    pkgconfig_2.0.1  
[46] xml2_1.1.1        rvg_0.1.5         lubridate_1.6.0   assertthat_0.2.0  httr_1.3.1       
[51] officer_0.1.6     R6_2.2.2          nlme_3.1-128
```

Screen shot out of browser (Rstudio) : 
![not_in_browser](https://user-images.githubusercontent.com/24737176/30634171-7e59652a-9dee-11e7-9e35-6b0b1d2bced8.png)

Screen shot in browser (I.E) : 
![in_browser](https://user-images.githubusercontent.com/24737176/30634166-79fc42fe-9dee-11e7-91e3-66f952240153.png)

**Comment by davidgohel** [comment]
I think you are using an old browser, right? Can you try with firefox, chrome or edge?

**Comment by antuki** [comment]
Yes it's working ! The answer was so simple ! Thank you !

**Comment by davidgohel** [comment]
What was your IE version?

**Comment by antuki** [comment]
Version : 11.0.9600.18449
Updates version : 11.0.35

**Comment by davidgohel** [comment]
I will try to fix that when I will understand what is going on in IE11

**Comment by antuki** [comment]
OK thank you !

**Comment by d-qn** [comment]
Hi, I experienced the same issue and I have just spent hours looking into it... I can confirm it is only an [IE issue](https://www.reddit.com/r/webdev/comments/4tf4ye/how_to_make_inline_svg_100_width_and_scaling_in_ie/).

My assumption is that it is due to ggiraph CSS changes wrapping the HTMLwidget/SVG, because I reverted to ggiraph 0.3.3 and that problem did not occur anymore (sorry I am css noob, I can't really say why the widget is no longer responsive in IE). 

I found a quick way to test it though
```
library(ggiraph)
library(htmltools)
dataset = iris
gg_point = ggplot(dataset, aes(x = Sepal.Length, y = Petal.Width,color = Species) ) +
	geom_point_interactive()
gg <- ggiraph(code = {print(gg_point)}, width = 1)
htmltools::save_html(gg, file = "test.html")
```
This will create an html file and the dependencies lib folder

**Comment by davidgohel** [comment]
I think that when I will be 80 years old, there will still have some issues with IE... 

No time to work on that but I will give a try next week.

**Comment by davidgohel** [comment]
Sorry for the delay.

It's not fully fixed. For now, if IE is been used, the graph is displayed with expected size but the *responsiveness* is turned off.

I will try to come back to ggiraph 3.3 behavior soon.

**Comment by d-qn** [comment]
Hi David,

sorry to get back on that issue. I used to be able to revert to ggiraph 3.3 to address this bug but now that _ggiraph_ supports _sf_ (awesome!), it is not really an option anymore. Anyway, FYI it seems indeed that on IE ggiraph 0.4.4 has the expected size but it is not left-aligned. I assume it is bc it generates by default a very wide svg. The problem with that behaviour is that it can't be shown in a narrowish window
* Illustration on IE of ggiraph 0.4.4 with a large browser window ![image](https://user-images.githubusercontent.com/4863520/43079527-a084618c-8e8d-11e8-8a8d-1f3e15e21ab7.png)
* Same but on a narrow window: 
![image](https://user-images.githubusercontent.com/4863520/43079579-c4962894-8e8d-11e8-82d1-9aa402b2ff89.png)

**Comment by davidgohel** [comment]
hi @d-qn, no problem

This is happening because IE does not handle SVG as it should do :( 

> [me, upper in this thread] It's not fully fixed. For now, if IE is been used, the graph is displayed with expected size but the responsiveness is turned off.

I set fixed size when IE is used: https://github.com/davidgohel/ggiraph/blob/master/inst/htmlwidgets/ggiraph.js#L56

This link is explaining better than I would what is the issue with IE: https://stackoverflow.com/questions/21922263/internet-explorer-not-honoring-svg-percentage-width/21946630#21946630

To summarise... size is set as *fixed* when browser is displaying the page. The alignment is centered (for all browsers). That's why you have this effect when using IE and resizing the window (in other browsers, it will be scaled regarding to the width of the window and stil centered) 

I am sorry I can't provide a solution for that issue at the moment. (I will keep on trying during following days).

**Comment by d-qn** [comment]
I see, bloody IE indeed. Thank you so much for looking into it! I am wondering how other htmlwidget generating svg handles that. If that could be helpful, I could see if there are HTMLwidgets that managed to address this IE issue.

**Comment by davidgohel** [comment]
Maybe, I tried but each has its own specific layout made of HTML, CSS, js... So I can not see clearly what should be done with ggiraph.

The obvious workaround is to dive into the old code and use it when IE is used. I found some solutions this morning (i.e. https://github.com/gavmck/scale-inline-svg-IE-polyfill/blob/master/index.js#L22), if I remember correctly, that was what I did in the old version. I will give a try.

**Comment by davidgohel** [comment]
Hi,

(sorry for the delay!)

I have fixed that in the new github version (soon on CRAN). The package does now support IE 10 and IE 11. There is no hope for IE < 10 :(

David
